### PR TITLE
chore: Add engine guards

### PR DESCRIPTION
**Pull request by bglossner**
Gives warning when using incompatible npm/node versions.

### CODE REVIEW

File: test-app/backend/package.json [hunk 1]
```diff
@@ -3,6 +3,10 @@
   "version": "1.0.0",
   "description": "Backend example app.",
   "main": "lib/server.js",
+  "engines": {
+    "node": ">=14.14.0",
```
Comment (rupaltotale) [inline test-app/backend/package.json:7]: I think the latest node is v18.7.0 (at least that's what mine got upgraded to). Should we use that here instead? Also, is there a way we can make these dynamic values (given that there are regular updates to node and npm?

File: test-app/backend/package.json [hunk 2]
```diff
@@ -3,6 +3,10 @@
...
+    "npm": ">=7"
```
Comment (rupaltotale) [inline test-app/backend/package.json:8]: I think latest is 8, not sure if it makes a difference though

**Review by rupaltotale — COMMENTED** [review]
thanks lol

File: test-app/backend/package.json (hunk 1 above)
Comment (bglossner) [reply test-app/backend/package.json:7]: I'm using v14.18 lol

**Review by rupaltotale — APPROVED** [review]

**Pull request merged**